core/src/common.cpp

```diff
@@ -207,6 +218,178 @@ namespace irods::http
 		return parse_url(fmt::format("http://ignored{}", _req.target()));
 	} // parse_url
 
+	auto url_encode_body(const body_arguments& _args) -> std::string
+	{
+		auto encode_pair{[](const body_arguments::value_type& i) {
+			return fmt::format("{}={}", encode(i.first), encode(i.second));
+		}};
+
+		return std::transform_reduce(
+			std::next(std::cbegin(_args)),
+			std::cend(_args),
+			encode_pair(*std::cbegin(_args)),
+			[](const auto& a, const auto& b) { return fmt::format("{}&{}", a, b); },
+			encode_pair);
+	}
+
+  auto safe_base64_encode(std::string_view _view) -> std::string {
+	constexpr auto char_per_byte_set{4};
+	constexpr auto byte_set_size{3};
+	auto max_size{char_per_byte_set * ((_view.size() + 2) / byte_set_size)};
+	auto max_size_plus_null_term{max_size + 1};
+		  
+	std::string encoded_data{};
```

Comment:
The curly braces are unnecessary. Please remove.

Reply:
Done.